Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs

```diff
@@ -90,7 +91,23 @@ public CosmosClientOptions()
         /// <remarks>
         /// Setting this property after sending any request won't have any effect.
         /// </remarks>
-        public string ApplicationName { get; set; }
+        public string ApplicationName
+        {
+            get => this.applicationName;
+            set
+            {
+                try
+                {
+                    NameValueHeaderValue nvhv = NameValueHeaderValue.Parse(value);
```

Comment:
nit:

```suggestion
                    NameValueHeaderValue.Parse(value);
```

The `nvhv` variable is not used